src/coreComponents/linearAlgebra/interfaces/hypre/HypreInterface.cpp

```diff
@@ -45,14 +45,23 @@ void HypreInterface::initialize()
                       "OMP_NUM_THREADS > 1 may not be optimal for certain hypre preconditioning options. " );
 #endif
 
-  HYPRE_Init();
+  HYPRE_Initialize();
 #if GEOS_USE_HYPRE_DEVICE == GEOS_USE_HYPRE_CUDA || GEOS_USE_HYPRE_DEVICE == GEOS_USE_HYPRE_HIP
   HYPRE_SetExecutionPolicy( HYPRE_EXEC_DEVICE );
   HYPRE_SetSpGemmUseVendor( 0 );
   HYPRE_DeviceInitialize();
 #endif
   HYPRE_SetMemoryLocation( hypre::memoryLocation );
   HYPRE_SetPrintErrorMode( 1 );
+
+#if defined(HYPRE_USING_UMPIRE)
+  HYPRE_SetUmpireUMPoolName( "HYPRE_UM" );
+  HYPRE_SetUmpireHostPoolName( "HYPRE_HOST" );
+  HYPRE_SetUmpireDevicePoolName( "HYPRE_DEVICE" );
+  HYPRE_SetUmpirePinnedPoolName( "HYPRE_PINNED" );
+#endif
+
+  HYPRE_SetLogLevel( getenv( "HYPRE_LOG_LEVEL" ) ? atoi( getenv( "HYPRE_LOG_LEVEL" ) ) : 0 );
```

Comment:
can it be set by linear solver log level?

Reply:
Yes, but I did this way because the information provided here is mainly for developer's. Other libraries such as umpire and rocsparse work this way as well (using env variable). I am open to change the strategy here if needeed

Reply:
sounds reasonable, thanks, no need to change

Reply:
I can remove this. What is the issue with env vars again?

Reply:
I personally don't like them. It's hard to keep track of who sets them. Also we don't really do it for anything else in GEOS.

Reply:
Here's a place where it's being used: https://github.com/GEOS-DEV/GEOS/blob/cd8175d5cef21a82a6558443ddb7f18600d5bb20/src/coreComponents/mainInterface/initialization.cpp#L141

This is an option supposed to be used by developers only. It's very convenient to turn it on/off if it's a env var wrt a XML input. I don't see any harm with using env var specifically, but I'm happy to remove it if that's the best solution